src/dsl.rs

```diff
@@ -10,27 +10,65 @@ use core::fmt::Debug;
 
 use self::cb::{Constraint, LookupBuilder};
 
+/// The **`CircuitContext`** struct is a generic structure designed to handle the context of a circuit for generic types **`F`**, **`TraceArgs`** and **`StepArgs`**. The struct contains a **`Circuit`** instance and implements methods to build the circuit, add various components, and manipulate the circuit.
+/// **Type Parameters**
+/// - **`F`**: A generic type representing the field of the circuit.
+/// - **`TraceArgs`**: A generic type representing the arguments passed to the trace function.
+/// - **`StepArgs`**: A generic type representing the arguments passed to the step_type_def function.
 pub struct CircuitContext<F, TraceArgs, StepArgs> {
     sc: Circuit<F, TraceArgs, StepArgs>,
 }
 
 impl<F, TraceArgs, StepArgs> CircuitContext<F, TraceArgs, StepArgs> {
+    /// # **Description:**
+    /// Adds a forward signal to the circuit with zero rotation and returns a **`Queriable`** instance.
+    /// # **Arguments:**
+    /// - **`name: &str`** - The name of the forward signal.
+    /// # **Return:**
+    /// - **`Queriable<F>`** - A **`Queriable`** instance representing the added forward signal.    
     pub fn forward(&mut self, name: &str) -> Queriable<F> {
         Queriable::Forward(self.sc.add_forward(name, 0), false)
     }
 
+    /// # **Description:**
+    /// Adds a forward signal to the circuit with the specified phase and returns a **`Queriable`** instance.
+    /// # **Arguments:**
+    /// - **`name: &str`** - The name of the forward signal.
+    /// - **`phase: usize`** - The phase of the forward signal.
+    /// # **Return:**
+    /// - **`Queriable<F>`** - A **`Queriable`** instance representing the added forward signal.
     pub fn forward_with_phase(&mut self, name: &str, phase: usize) -> Queriable<F> {
         Queriable::Forward(self.sc.add_forward(name, phase), false)
     }
 
+    /// # **Description:**
+    /// Imports a halo2 advice column into the circuit and returns a **`Queriable`** instance.
+    /// # **Arguments:**
+    /// - **`name: &str`** - The name of the halo2 advice column.
+    /// - **`column: Halo2Column<Advice>`** - The advice column to import.
+    /// # **Return:**
+    /// - **`Queriable<F>`** - A **`Queriable`** instance representing the imported halo2 advice column.
     pub fn import_halo2_advice(&mut self, name: &str, column: Halo2Column<Advice>) -> Queriable<F> {
         Queriable::Halo2AdviceQuery(self.sc.add_halo2_advice(name, column), 0)
     }
 
+    /// # **Description:**
+    /// Imports a halo2 fixed column into the circuit and returns a **`Queriable`** instance.
+    /// # **Arguments:**
+    /// - **`name: &str`** - The name of the halo2 fixed column.
+    /// - **`column: Halo2Column<Fixed>`** - The fixed column to import.
+    /// # **Return:**
+    /// - **`Queriable<F>`** - A **`Queriable`** instance representing the imported halo2 fixed column.
```

Comment:
We should leave the main sentence in the first line. The document treats it specially. For example, when you search the function, the doc shows the first line. And if we put "Description" in the first line, the doc shows something unhelpful

![Screenshot 2023-05-03 at 2 53 59 PM](https://user-images.githubusercontent.com/3391420/235849746-fc2b8610-b64d-4f5b-ae5a-1ff19ae86361.png)

We don't need to doc the params and the return types because we can jump and search them in the document.
```suggestion
    /// Imports a halo2 fixed column into the circuit and returns a **`Queriable`** instance.
```

Reply:
fixed in pr#18